### PR TITLE
optimize sparse softmax cross entropy gpu

### PR DESCRIPTION
**Pull request by guo-ran**

### CODE REVIEW

File: oneflow/user/kernels/sparse_softmax_cross_entropy_kernel.cu [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                                 reinterpret_cast<half*>(prob));
+}
+
+}  // namespace
+
+template<typename T, typename K>
+class SparseSoftmaxCrossEntropyKernel final : public user_op::OpKernel {
+ public:
+  SparseSoftmaxCrossEntropyKernel() = default;
+  ~SparseSoftmaxCrossEntropyKernel() = default;
```
Comment (liujuncheng) [inline oneflow/user/kernels/sparse_softmax_cross_entropy_kernel.cu:43]: `override`

**Review by liujuncheng — APPROVED** [review]
